sequencer.c

```diff
@@ -861,21 +879,7 @@ static int run_git_commit(const char *defmsg, struct replay_opts *opts,
 	if (opts->allow_empty_message)
 		argv_array_push(&cmd.args, "--allow-empty-message");
 
-	if (cmd.err == -1) {
-		/* hide stderr on success */
-		struct strbuf buf = STRBUF_INIT;
-		int rc = pipe_command(&cmd,
-				      NULL, 0,
-				      /* stdout is already redirected */
-				      NULL, 0,
-				      &buf, 0);
-		if (rc)
-			fputs(buf.buf, stderr);
-		strbuf_release(&buf);
-		return rc;
-	}
-
-	return run_command(&cmd);
+	return run_command_silent(&cmd, is_rebase_i(opts) && !(flags & EDIT_MSG));
```

Comment:
Another possibility is to remove the silent argument from run_command_silent() and to have:

```
if (is_rebase_i(opts) && !(flags & EDIT_MSG))
        return run_command_silent(&cmd);
else
        return run_command(&cmd);
```

Reply:
True. It’s probably less confusing.